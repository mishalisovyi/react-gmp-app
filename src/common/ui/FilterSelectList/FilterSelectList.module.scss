@use '../../../colors';
@use '../../../mixins';

.FilterSelectList {
  display: flex;
  align-items: center;
  column-gap: 32px;

  &__label {
    @include mixins.typography-button-small-lightened;

    color: rgba(colors.$white, .6);
  }

  &__control, &__control:focus {
    @include mixins.typography-button-small;

    color: colors.$white;
    background-color: colors.$transparent;
    border: none;
    cursor: pointer;

    .control {
      &__option {
        background-color: colors.$young-night;
      }
    }
  }
}