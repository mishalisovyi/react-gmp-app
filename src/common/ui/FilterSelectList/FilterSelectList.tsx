import React, { ChangeEvent } from 'react';
import { SelectListValue, SelectOptionDataItem } from 'common/interfaces';
import styles from './FilterSelectList.module.scss';

interface FilterSelectListProps {
  label: string;
  options: SelectOptionDataItem[];
  defaultValue: SelectListValue
  onSelectionChanged: (item: string) => void;
}

export function FilterSelectList({
  label,
  options,
  defaultValue,
  onSelectionChanged,
}: FilterSelectListProps) {
  const onSelectValueChanged = ({ target: { value } }: ChangeEvent<HTMLSelectElement>) => {
    onSelectionChanged(value);
  };

  return (
    <div className={styles['FilterSelectList']}>
      <label className={styles['FilterSelectList__label']}>{label}</label>

      <select className={styles['FilterSelectList__control']} defaultValue={defaultValue} onChange={onSelectValueChanged}>
        {options.map((item) => {
          return <option className={styles['control__option']} key={`${item.title}_${item.value}`} value={item.value}>{item.title}</option>;
        })}
      </select>
    </div>
  );
}
