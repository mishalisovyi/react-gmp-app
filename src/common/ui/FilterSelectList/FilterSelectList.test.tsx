import React from 'react';
import renderer from 'react-test-renderer';

import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { FilterSelectList } from './FilterSelectList';

const selectListOptions = [{
  title: 'Test 1',
  value: 'test 1',
}, {
  title: 'Test 2',
  value: 'test 2',
}];

const selectionChangedHandler = jest.fn();

it('should render select list', () => {
  const tree = renderer.create(
    <FilterSelectList label="test" options={selectListOptions} defaultValue={selectListOptions[0].value} onSelectionChanged={selectionChangedHandler} />,
  );

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should handle option selection', async () => {
  const selectedOptionValue = selectListOptions[1].value;

  const { container } = render(
    <FilterSelectList label="test" options={selectListOptions} defaultValue={selectListOptions[0].value} onSelectionChanged={selectionChangedHandler} />,
  );

  const [select] = container.getElementsByTagName('select');

  await userEvent.selectOptions(select, selectedOptionValue);

  expect(selectionChangedHandler).toHaveBeenCalledWith(selectedOptionValue);
});
