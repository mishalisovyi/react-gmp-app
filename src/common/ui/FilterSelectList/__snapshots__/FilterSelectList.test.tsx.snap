// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render select list 1`] = `
<div>
  <label>
    test
  </label>
  <select
    defaultValue="test 1"
    onChange={[Function]}
  >
    <option
      value="test 1"
    >
      Test 1
    </option>
    <option
      value="test 2"
    >
      Test 2
    </option>
  </select>
</div>
`;
